builtin/logical/pki/path_issue_sign.go

```diff
@@ -429,6 +429,49 @@ func (b *backend) pathIssueSignCert(ctx context.Context, req *logical.Request, d
 	return resp, nil
 }
 
+type caChainOutput struct {
+	chain []*certutil.CertBlock
+}
+
+func newCaChainOutput(parsedBundle *certutil.ParsedCertBundle, data *framework.FieldData) caChainOutput {
+	if filterCaChain := data.Get("remove_roots_from_chain").(bool); filterCaChain {
+		var myChain []*certutil.CertBlock
+		for _, certBlock := range parsedBundle.CAChain {
+			cert := certBlock.Certificate
+			if (len(cert.AuthorityKeyId) > 0 && !bytes.Equal(cert.AuthorityKeyId, cert.SubjectKeyId)) ||
+				(len(cert.AuthorityKeyId) == 0 && !bytes.Equal(cert.RawIssuer, cert.RawSubject)) {
```

Comment:
While this code was cribbed from certutil and I'm thus fine leaving it, it'd be good to also verify `cert.CheckSignatureFrom(cert) == nil`, i.e., that the cert is truly both self-referencing and has a valid self-signature.

With fancy chains (e.g., a rotation of the same CA cert under new key material), generated outside of vault (no AKID), you can run into the case where the issuers will line up, and you probably want to include the cross-bridge but not either self-signed roots. This will elide the cross-bridge.